src/EditModel/event-program/EventProgramStepperContent.js

```diff
@@ -10,26 +10,46 @@ import AssignDataElements from './assign-data-elements/AssignDataElements';
 import EditDataEntryForm from './create-data-entry-form/CreateDataEntryForm.component';
 import AssignOrganisationUnits from './assign-organisation-units/AssignOrganisationUnits';
 import EventProgramNotifications from './notifications/EventProgramNotifications';
-import { createFormFor } from '../formHelpers';
+import { createFieldConfigsFor } from '../formHelpers';
 import { editFieldChanged } from './actions';
-import { flattenRouterProps, wrapInPaper } from '../componentHelpers';
+import { wrapInPaper } from '../componentHelpers';
 import fieldOrder from '../../config/field-config/field-order';
+import FormBuilder from 'd2-ui/lib/forms/FormBuilder.component';
+import {editProgramStageField} from "./tracker-program/program-stages/actions";
 
-const stepperConfig = () => {
-    const program$ = eventProgramStore.map(get('program'));
+const eventProgramFields = fieldOrder.for('eventProgram');
+const eventProgramStageFields = fieldOrder.for('eventProgramStage');
+let ProgramForm = props => (
+    <FormBuilder
+        fields={props.fieldConfigs}
+        onUpdateField={props.editFieldChanged}>
+    </FormBuilder>)
+
+ProgramForm = createFieldConfigsFor('program', eventProgramFields, undefined, null, true, 'eventProgram')(ProgramForm);
 
-    const mapDispatchToProps = dispatch =>
-        bindActionCreators({ editFieldChanged }, dispatch);
+let ProgramStageForm = props => (
+    <FormBuilder
+        fields={props.fieldConfigs}
+        onUpdateField={props.editProgramStageFieldChange}>
+    </FormBuilder>
+)
+ProgramStageForm = createFieldConfigsFor('programStage', eventProgramStageFields, undefined, null, true, 'eventProgramStage')(ProgramStageForm);
```

Comment:
When there are so many parameters we should consider using a config object instead.

Reply:
I've thought about it, but I did not want to change it now as it's used all over the place.